irr/src/CIrrDeviceLinux.cpp

```diff
@@ -862,10 +862,19 @@ bool CIrrDeviceLinux::run()
 
 				irrevent.EventType = irr::EET_KEY_INPUT_EVENT;
 				irrevent.KeyInput.PressedDown = false;
-				irrevent.KeyInput.Char = 0; // on release that's undefined
 				irrevent.KeyInput.Control = (event.xkey.state & ControlMask) != 0;
 				irrevent.KeyInput.Shift = (event.xkey.state & ShiftMask) != 0;
 				irrevent.KeyInput.Key = getKeyCode(event);
+				{ // Pass a char on release as well since this is needed for the scancode stub.
```

Comment:
> This fragment seems a bit weird to me:

TL;DR: it is weird.

As for where this came from: this is actually copied from around ~50 lines further down the source file (and apparently was around before IrrlichtMT was fored). I copied this under the impression that it "just works" without seriously looking into this too carefully. Turns out this is a huge WTF.

> In the call to `XLookupString`, `tmp.buf` and `sizeof(tmp.buf)` are passed. The man page indicates to me that `XLookupString` will write a couple characters there. But afterwards, a different union field, `wbuf` is accessed. [...]
> 
> Is just converting to `wchar_t` in this manner correct? Doesn't this assume that an UTF-32 encoding is used? 

_Ideally_ we would use `XmbLookupString` or `XwcLookupString` here, in which case we would use `mbtowc` (or something similar), but these two [should only be used in `KeyPress` events](https://linux.die.net/man/3/xwclookupstring), so that becomes irrelevant here.

So we use `XLookupString`, which actually returns the string [in the ISO-8859-1 encoding](https://linux.die.net/man/3/xlookupstring), which is compatible with Unicode in the 0x80-0xFF range, so IMO a direct cast to `wchar_t` should be fine.

(By the way, I originally copied only this snippet because I thought that the input context would not be relevant here, which is apparently different thing.)

> Doesn't this make endianness assumptions?

Apparently yes. It seems like nobody so far has noticed this because it requires a setup with a big endian machine _and_ (until now) where `XInputContext` is not available.

> Furthermore, the return value of `XLookupString`, the number of characters written to `buf`, is ignored. Arguably this is probably fine if the endianness checks out because the rest has been zero-initialized, but again not quite obvious to the reader.

IMO this is (aside from the endianness) not really an issue because we only need the first character (see below).

> Also, why can / do we want to ignore `wbuf[1]`?

I don't think we have any reason to consider the present of a second character. Irrlicht's keypress/keyrelease events only use one character.

---

I think I will change this to `irrevent.KeyInput.Char = buf[0]` (to address the endianness issue) as I don't think there is anything else I can do about this mess.

Reply:
Resolving this since I have reverted changes to non-SDL `IrrlichtDevice`s.

(Note that the issue mentioned here is still present in the existing code, but fixing that is outside the scope of this PR.)